docs/topics/build-tools-api.md

```diff
@@ -0,0 +1,104 @@
+[//]: # (title: Build tools API)
+
+<primary-label ref="experimental-general"/>
+
+<!-- Add diagram -->
+
+Kotlin 2.2.0 introduces an experimental Build tools API (BTA) that simplifies how build systems integrate with the 
+Kotlin compiler.
+
+Traditionally, adding full Kotlin support to a build system (like incremental compilation, Kotlin compiler plugins, 
+daemons, and Kotlin Multiplatform) required significant effort. The BTA aims to reduce this complexity by providing
+a unified API between build systems and the Kotlin compiler ecosystem.
+
+The BTA defines a single entry point that build systems can implement. This removes the need to deeply integrate with internal compiler details.
+
+Currently, the BTA supports Kotlin/JVM **only**.
+
+> The BTA itself is not yet publicly available for direct use in your own build tool integrations.
+> If you're interested in the proposal or want to share feedback, see the [KEEP](https://github.com/Kotlin/KEEP/issues/421).
```

Comment:
@ALikhachev should we also link here BTA implementation issue: https://youtrack.jetbrains.com/issue/KT-76255/Implement-Build-Tools-API? 

So interested users could track progress of it?